File: tests/cases/reference/gate/csharp/inputs/src/Catalog/Author.cs
using System.Collections.Generic;

namespace Catalog;

public class Author
{
    public string Name { get; }
    public float TestVar { get; }

    public Author(string name) { Name = name; TestVar = 0; }

    // public -> private: an access-level change.
    private List<Book> Books() { return new List<Book>(); }

    // instance -> static: a modifier change.
    public static string Format() { return "x"; }
}

File: examples/python_demo/catalog/author.py
"""Authors in the bookstore catalog."""


class Author:
    def __init__(self, name: str, country: str = "Unknown") -> None:
        self.name: str = name
        self.country: str = country
        self.test_var: float = 1.0

    def display_name(self) -> str:
        return f"{self.name} ({self.country})"
